### PR TITLE
TLI 101.b -- Link up API for ml_model for multiple models to Dashboard 2

### PR DESCRIPTION
**Pull request by Jai0212**
## Description:

This pull request introduces functionality to allow users to upload machine learning models via the Dashboard 2 interface. The backend will analyze the uploaded models and return the corresponding output for further visualization.

Additionally, error handling has been implemented to ensure that users are informed if they upload unsupported or incorrect file formats.

## Key Changes:
Model Upload: Users can now upload machine learning models through the Dashboard 2 interface.
Model Analysis: Once the model is uploaded, the backend will analyze the model and return the relevant output.
Error Handling: Proper error handling is in place to notify the user if an unsupported or incorrect model file is uploaded.

## Acceptance Criteria:
[] Dashboard 2 should be fully functional with proper integration of graph displays showing model analysis.
[] Error handling should be implemented to guide the user when they upload incorrect files.

## Testing Notes:
Upload valid model files to ensure that the backend performs the correct analysis and returns the expected results.
Try uploading invalid or incorrect files to verify that appropriate error messages are displayed.

image

### CODE REVIEW

**Comment by armaganngul** [comment]
Looks good! @tvirat 's implementation of the background for dashboard2 isnt here though? did you not pull?

File: backend/app/controllers/app.py [hunk 1]
```diff
@@ -251,7 +252,19 @@ def generate_for_all_models():
...
             models[i] = os.path.join(UPLOAD_FOLDER, curr_user, models[i])
-        print(models)
+        print("Multiple Models Paths:", models)
+
+        evaluator = EvaluateModelsUseCase(models)
+        output = evaluator.execute()
+
+        print("Multiple Models Output", output)
+
+        output_filtered = []
```
Comment (armaganngul) [inline backend/app/controllers/app.py:262]: is output_filtered the elements of the dictionary?

File: backend/demo.py [hunk 2]
```diff
@@ -47,15 +47,15 @@
...
-for i in result:
-    print(
-        i.get_feature1(),
-        i.get_feature2(),
-        i.get_accuracy(),
-        i.get_false_positive_rate(),
-        i.get_false_negative_rate(),
-        i.get_combination_label(),
-    )
+# generate = Generate(file_repo, db_repo)
```
Comment (armaganngul) [inline backend/demo.py:50]: If this is gonna be commented out it might be good to delete it!

File: backend/uploads/jj@gmail.com/model_with_score_2.pkl
Comment (armaganngul) [inline backend/uploads/jj@gmail.com/model_with_score_2.pkl:1]: this is just a placeholder model right?

**Review by armaganngul — COMMENTED** [review]
Not really crucial but some comments and questions @Jai0212

File: backend/demo.py (hunk 2 above)
Comment (Jai0212) [reply backend/demo.py:50]: I will be deleting this entire file when we finish the project as this is for testing purposes

File: backend/uploads/jj@gmail.com/model_with_score_2.pkl
Comment (Jai0212) [reply backend/uploads/jj@gmail.com/model_with_score_2.pkl:1]: Yea, we need to develop more for better testing

File: backend/app/controllers/app.py (hunk 1 above)
Comment (Jai0212) [reply backend/app/controllers/app.py:262]: The return statement returns a dictionary with keys but Frotnend expects a list of simple dictionaries. So I had to filter out the key values for the correct JSON return

**Comment by Jai0212** [comment]
> Looks good! @tvirat 's implementation of the background for dashboard2 isnt here though? did you not pull?

He hasn't pushed it yet as there are some changes to be made in Dashboard 1

**Pull request merged**